src/Scripting/CoreTestUtilities/README.md

```diff
@@ -0,0 +1,15 @@
+# Scripting Unit Tests
+
+## Native Memory Pressure
+
+The scripting engine in Roslyn makes heavy use of creating `MetadatImageReference` over a `Stream`. That is the way in which runtime assembly references are added. This form of `MetadataReference` results in a lot of native allocations as opposed to using an `ImmutableArray<byte>` which has none. By default these native allocations are collected by the finalizer as `MetadadataImageReference` is not disposable.
+
```

Comment:
Have we considered fixing this by using `ImmutableArray<byte>` instead of relying on `MetadataImageReference` over `Stream`?

Reply:
I _believe_ the reason scripting uses `Stream` is that in the normal use case the belief is that it should result in less overall allocations. The compiler lazily loads data from the PE file so in theory you could load a lot of assemblies with only targeted allocations. As opposed to the `ImmutableArray<byte>` case where you have to eagerly load them all into memory. 

The unit tests are just taking the downsides of this approach to the extreme.